### PR TITLE
Enforce non-nullable fields for BQ dialect

### PR DESCRIPTION
**Pull request by MichaelSchmidt82**
TL;DR These changes remove the optional `NULL` fields for BigQuery in `replacementPatterns.csv`.  This is a fix for #265 and includes changes for similar types that are causing issues.  Added a test.

There are several fields for BigQuery that will produce a syntax error if they are nullable.  The changes here enforce that they mustn't be null.  When you use Postgres as your flavor of choice the word ["offset" is reserved](https://www.postgresql.org/docs/8.1/sql-keywords-appendix.html).  But in BQ land, this word isn't, and using it as a literal (aka `"offset"` vs `offset`) produces a syntax error.  A test is added for this case.

However, it is unfortunate that OHDSI/CommonDataModel has a [hardcoded `if` branch](https://github.com/OHDSI/CommonDataModel/blob/af1effdedc300500555fe66e68b118b9f6f844bf/R/createDdl.R#L77) that will apply the double quotes specifically to this word.  It would be ideal to remove this logic and place it here instead.

### CODE REVIEW

**Review by schuemie — APPROVED** [review]

**Comment by schuemie** [comment]
The related [issue](https://github.com/OHDSI/SqlRender/issues/265) specifically mentioned `numeric`. Any reason you did not add that data type to this pull request?

**Comment by MichaelSchmidt82** [comment]
> The related [issue](https://github.com/OHDSI/SqlRender/issues/265) specifically mentioned `numeric`. Any reason you did not add that data type to this pull request?

[these lines](https://github.com/MichaelSchmidt82/SqlRender/blob/782689c3bb5a4667105842a5d119a0a0a81bda98/inst/csv/replacementPatterns.csv#L680) add other data types.

**Comment by schuemie** [comment]
> [these lines](https://github.com/MichaelSchmidt82/SqlRender/blob/782689c3bb5a4667105842a5d119a0a0a81bda98/inst/csv/replacementPatterns.csv#L680) add other data types.

Yes, that is my point. It adds other data types, not the one mentioned in the original issue. In other words, even after this pull request, 
```sql
CREATE TABLE my_table (x NUMERIC NULL);
```
will be translated incorrectly. Is there a reason for that that I'm missing?

**Review by schuemie — APPROVED** [review]
